### DISCUSSION TITLE
Inquiry about CartesianChartModelProducer benefits

### DISCUSSION THREAD

**Discussion by siyingch86** [patrykandpatrick/vico#611] [Questions] [answered]
"We recommend using [CartesianChartModelProducer]because it offers performance benefits and animations."
Could you provide more details about how this offers performance benefits behind the scene? Any data support this evidence? Thanks!

**Comment by patrickmichalik** [comment, marked as answer]
Hello! `CartesianChartModelProducer` moves a significant amount of work off the main thread. It asynchronously creates `CartesianChartModel`s and their `CartesianLayerModel`s, which is important because the calculations that go into this can be quite complex. `ChartValues` updates (including `AxisValueOverrider` calls) are also asynchronous. If these operations were to be performed on the main thread, they could cause FPS drops. With `CartesianChartModelProducer`, however long they take, the UI’s smoothness is unaffected. `CartesianChartModelProducer` also asynchronously creates `DrawingModel`s, performing certain drawing-related calculations in advance and decreasing the number of operations that the drawing logic has to do.

I’m afraid we don’t have any recent comparative performance measurements that I could share, but I hope the explanation above clears things up.

**Reply by siyingch86** [reply]
Thanks for detailed reply!

**Reply by patrickmichalik** [reply]
You’re welcome!
